src/doc.c

```diff
@@ -2128,21 +2134,73 @@ TypeFunction *isTypeFunction(Dsymbol *s)
 /****************************************************
  */
 
+Parameter *isFunctionParameter(Dsymbol *s, const utf8_t *p, size_t len)
+{
+    TypeFunction *tf = isTypeFunction(s);
+    if (tf && tf->parameters)
+    {
+        for (size_t k = 0; k < tf->parameters->dim; k++)
+        {
+            Parameter *fparam = (*tf->parameters)[k];
+            if (fparam->ident && cmp(fparam->ident->toChars(), p, len) == 0)
+            {
+                return fparam;
+            }
+        }
+    }
+    return NULL;
+}
+
+/****************************************************
+ */
+
 Parameter *isFunctionParameter(Dsymbols *a, const utf8_t *p, size_t len)
 {
     for (size_t i = 0; i < a->dim; i++)
     {
-        TypeFunction *tf = isTypeFunction((*a)[i]);
-        if (tf && tf->parameters)
+        Parameter *fparam = isFunctionParameter((*a)[i], p, len);
+        if (fparam)
         {
-            for (size_t k = 0; k < tf->parameters->dim; k++)
+            return fparam;
+        }
+    }
+    return NULL;
+}
+
+/****************************************************
+ */
+
+Parameter *isEponymousFunctionParameter(Dsymbols *a, const utf8_t *p, size_t len)
+{
+    for (size_t i = 0; i < a->dim; i++)
+    {
+        TemplateDeclaration* td = (*a)[i]->isTemplateDeclaration();
+        if (td && td->onemember)
+        {
+            // Case 1: we refer to a template declaration inside the template
+            td = td->onemember->isTemplateDeclaration();
+        }
+        if (!td)
+        {
+            // Case 2: we're an alias to a template declaration
+            AliasDeclaration *ad = (*a)[i]->isAliasDeclaration();
+            if (ad && ad->aliassym)
+            {
+                td = ad->aliassym->isTemplateDeclaration();
+            }
```

Comment:
I think this 'case 2' is unnecessary so it would cause incorrect warning suppression.

See I posted test case: https://issues.dlang.org/show_bug.cgi?id=14633#c2